projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/deploy/JobImageBuilderTest.java

```diff
@@ -232,4 +236,140 @@ public void buildImage_jobFailed_failure()
             true);
     Assertions.assertFalse(result);
   }
+
+  @Test
+  public void
+      buildImage_deploymentDataJobBaseImageNullAndSupportedPythonVersions_shouldCreateCronjobUsingSupportedPythonVersions()
+          throws InterruptedException, ApiException, IOException {
+    // Set base image property to null
+    ReflectionTestUtils.setField(jobImageBuilder, "deploymentDataJobBaseImage", null);
+
+    when(dockerRegistryService.builderImage()).thenReturn(TEST_BUILDER_IMAGE_NAME);
+    when(kubernetesService.listJobs()).thenReturn(Collections.emptySet());
+    var builderJobResult =
+        new KubernetesService.JobStatusCondition(true, "type", "test-reason", "test-message", 0);
+    when(kubernetesService.watchJob(any(), anyInt(), any())).thenReturn(builderJobResult);
+    when(supportedPythonVersions.getJobBaseImage("3.11")).thenReturn("test-base-image");
+
+    JobDeployment jobDeployment = new JobDeployment();
+    jobDeployment.setDataJobName(TEST_JOB_NAME);
+    jobDeployment.setGitCommitSha("test-commit");
+    jobDeployment.setEnabled(true);
+    jobDeployment.setPythonVersion("3.11");
+
+    ArgumentCaptor<Map<String, String>> captor = ArgumentCaptor.forClass(Map.class);
+
+    var result = jobImageBuilder.buildImage("test-image", testDataJob, jobDeployment, true);
+
+    verify(kubernetesService)
+        .createJob(
+            eq(TEST_BUILDER_JOB_NAME),
+            eq(TEST_BUILDER_IMAGE_NAME),
+            eq(false),
+            eq(false),
+            captor.capture(),
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            anyLong(),
+            anyLong(),
+            anyLong(),
+            any(),
+            any());
+
+    Map<String, String> capturedEnvs = captor.getValue();
+    Assertions.assertEquals("test-base-image", capturedEnvs.get("BASE_IMAGE"));
+
+    verify(kubernetesService).deleteJob(TEST_BUILDER_JOB_NAME);
+    Assertions.assertTrue(result);
+  }
+
+  @Test
+  public void buildImage_shouldCreateCronjobUsingDeploymentDataJobBaseImage()
```

Comment:
nit: follow the same naming convention as the other ones.

Reply:
Renamed, thanks